### PR TITLE
Switch AFN simulation control from ints to an enum

### PR DESCRIPTION
**Pull request by jasondegraw**
Pull request overview
---------------------
The type of AirflowNetwork simulation (MultizoneWithDistribution, etc.) was being controlled by an integer variable, this PR changes that into an enumeration. There were some areas where things were a bit unclear due to the greater than/less than logic, hopefully those are all sorted out now. In the course of fixing a problem in the unit tests, I found that one of the unit test source files had gotten too big for GitHub to display, so I've reorganized the AFN tests (mostly) so all the files should display. The biggest one (AirflowNetworkHVAC.unit.cc) is still close to the borderline.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp:237]: These were the integer values that we want to replace with an enumeration.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp:392]: Moved this to Solver.hpp where it makes more sense.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp:84]: Here's the replacement enumeration. I've kept some of the original information around for reference, we can delete some of this later.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp]: This is the new simulation control object, there's still more to do here, but can tackle that in a later PR.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp:501]: Some items from the old simulation control object make more sense out here, these values are changing and pretty much everything in the simulation control object should not.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:2034]: I've added a couple of convenience booleans to simplify some of the logic. This can be inconvenient if one forgets to add them to the clear_state functions (oops) but since these were getting checked a lot it makes sense to use a single boolean for readability.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:2036]: And here is how one of the convenience booleans get used. Rather than having to look at what is being checked to see what is intended, the variable name is all that needs to be understood.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:2246]: I'm still using the IDF input style here, this all gets removed in the input refactor.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:8733]: Some of the logic doesn't look too much better, but it is at least a little shorter.

**Comment by jasondegraw** [comment]
As far as I know, this one is ready to go. Of the int/enum changes that need to be made in AFN, this one is the most involved because so many of the other modules are checking the control type. That's why I did this one separately.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp]: These integer value comments should be removed to avoid confusion, because I think they are just zero based in the enum now.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp:677]: If possible, I'd still use the `*this` expression to clean out this clear_state.  It's super easy to forget to update an assignment in these functions.  It's not necessary on this PR though.

**Review by Myoldmopar — COMMENTED** [review]
Nice cleanup here.  I think the only thing that should be cleaned before merge is deleting the commented integers (unless I'm missing something and they should stay).

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp]: I think I have mainly reproduced the information in the comments on the enum, so it's not a big deal either way.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Solver.hpp:677]: I understand what you're saying, but I'm still not sure about assigning to `*this` in that way. I did look at it a little in compiler explorer, and gcc and clang are generating move `operator=` and other things that make me a little uncomfortable.

**Comment by jasondegraw** [comment]
@Myoldmopar Let's hold this until after #9465, I'll deal with any conflicts.

**Comment by Myoldmopar** [comment]
@jasondegraw are you expecting the table diff that popped up in the last round of CI?  Might be due to develop moving, not sure, just asking before I dig in to check on them.  Thanks.

**Comment by jasondegraw** [comment]
@Myoldmopar I'm not seeing a table diff, maybe I'm not looking in the right place. Where are you seeing it?

**Comment by Myoldmopar** [comment]
Good catch!  For some reason, when I am viewing the PR, the last CI results that are showing up are for [a0f2f52](https://github.com/NREL/EnergyPlus/pull/9449/commits/a0f2f523d02f570852924bb7a92c8677a5d20016).  When I click on the individual last commit [25005da](https://github.com/NREL/EnergyPlus/pull/9449/commits/25005daa2e24d886a5c499603028fb2fdca4219d) I see that it is all clean.  Carrying on...  

Anything else here anyone?  I'll take one more look at the changes but I am guessing this is ready to drop shortly.

**Review by Myoldmopar — APPROVED** [review]
Looks great to me, and hey a reduction of 87 lines along with all the improved patterns.  I'm good with this going in.

**Comment by jasondegraw** [comment]
@Myoldmopar No problem, that has happened to me too, and I wasn't sure that it wasn't happening to me when I got your message. I've taken a quick run back through and I think this is good to go.

**Comment by Myoldmopar** [comment]
Agreed, it looks great.  @amirroth are you actively reviewing the changes?  I'll hold off if you are, otherwise I think this is merging.

Aside:  It would be nice if you could flip a switch on a PR that says "I'm looking this over right now" and it would expire in like an hour if you don't complete.  That way others would know you are actively looking it over.

**Comment by amirroth** [comment]
> Agreed, it looks great. @amirroth are you actively reviewing the changes? I'll hold off if you are, otherwise I think this is merging.
> 
> Aside: It would be nice if you could flip a switch on a PR that says "I'm looking this over right now" and it would expire in like an hour if you don't complete. That way others would know you are actively looking it over.

I am not. Just a drive-by comment. I guess it makes lines a bit longer, but I find snake_case much easier to read because it doesn't require mental effort to find word breaks. And too many capital letters make it look like the code is yelling at you.

**Pull request merged**
